src/main/java/htsjdk/samtools/CRAMFileReader.java

```diff
@@ -214,8 +218,25 @@ public CRAMFileReader(final InputStream stream,
     public CRAMFileReader(final File cramFile,
                           final File indexFile, final CRAMReferenceSource referenceSource,
                           final ValidationStringency validationStringency) throws IOException {
-        this(new FileInputStream(cramFile), indexFile, referenceSource, validationStringency);
+        this(new FileInputStream(cramFile),
+                wrapIndexFile(cramFile, indexFile),
```

Comment:
Rather than having this somewhat redundant single-use static method, I think it would be cleaner to remove it, and instead factor out the common initialization code from the constructor it's delegating to, and call that method from both constructors (this method would no longer delegate to the other one).